src/components/Modals/ModalKacy/index.tsx

```diff
@@ -111,7 +111,7 @@ const ModalKacy = () => {
     }
 
     if (wallet?.provider) {
-      getKacyBalanceInWallet(wallet.accounts[0].address)
+      getKacyBalanceInWallet(wallet?.accounts[0]?.address)
```

Comment:
No need to use optional chaining
```suggestion
      getKacyBalanceInWallet(wallet.accounts[0]?.address)
```